src/instructlab/config/init.py

```diff
@@ -327,7 +345,11 @@ def get_gpu_or_cpu() -> tuple[str, bool, bool, int, int]:
     is_cpu = False
     is_linux = False
     # try nvidia
-    if torch.cuda.is_available() and torch.version.hip is None:
+    if (
```

Comment:
nit: No issue merging this change since it is already on main.  But would it be better to check for `torch.version.hip is None and not is_hpu_available()` above this if and the one below.

Reply:
I could open a follow up and we could backport both if we want. 

This would allow me to use the mocked calls in the test as well and assert each is only called once.

Reply:
I don't think there is a need to backport in this case since there isn't a functional difference.  +1 to changing on main if you agree it's a good change.